service/controller/v5/resource/blobobject/update.go

```diff
@@ -0,0 +1,117 @@
+package blobobject
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/azure-operator/service/controller/v5/blobclient"
+	"github.com/giantswarm/azure-operator/service/controller/v5/key"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller"
+)
+
+func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	containerObjectToUpdate, err := toContainerObjectState(updateChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	storageAccountsClient, err := r.getAccountsClient()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var blobClient *blobclient.BlobClient
+	{
+		c := blobclient.Config{
+			ContainerName:         key.BlobContainerName(),
+			GroupName:             key.ClusterID(customObject),
+			StorageAccountName:    key.StorageAccountName(customObject),
+			StorageAccountsClient: storageAccountsClient,
+		}
+
+		blobClient, err = blobclient.New(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	err = blobClient.Boot(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for key, containerObject := range containerObjectToUpdate {
+		if containerObject.Key != "" {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating container object %#q", key))
+
+			_, err := blobClient.PutBlockBlob(ctx, key, containerObject.Body)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created container object %#q", key))
+		} else {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not create container object %#q", key))
+			r.logger.LogCtx(ctx, "level", "debug", "message", "container object already exists")
+		}
+	}
+
+	return nil
+}
+
+func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*controller.Patch, error) {
+	create, err := r.newCreateChange(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	update, err := r.newUpdateChange(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	patch := controller.NewPatch()
+	patch.SetCreateChange(create)
+	patch.SetUpdateChange(update)
+
+	return patch, nil
+}
+
+func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	currentContainerObject, err := toContainerObjectState(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredContainerObject, err := toContainerObjectState(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if the container objects should be updated")
+
+	updateState := map[string]ContainerObjectState{}
```

Comment:
I don't like this is a map. There is an assumption that `ContainerObjectState.Key` is equal to the map key. I would rather use slice and search for the object based on its `Key` property. Performance isn't really crucial here.

Reply:
Frankly speaking I didn't get this :/

Reply:
I think updateState type should be `[]ContainerObjectState` instead of the map or `Key` should be removed from `ContainerObjectState`. I like the first more approach more.

Reply:
Ok, I got this. But with the lists, we then need nested loops for comparing current/desired states. Is that a nicer approach?